proj/src/main/webapp/chat.js

```diff
@@ -634,6 +639,39 @@ function friendRequestButton(uid) {
     });
 }
 
+function blockButton(uid) {
+    const currUserRef = firebase.database().ref('/users/'+currentUID+'/blocked/'+uid);
+    const button = document.getElementById('block');
+
+    currUserRef.off();
+    currUserRef.on('value', function(snap) {
+        if (snap.val()) {
+            // user is blocked
+            button.innerText = 'unblock user';
+            button.onclick = function() {
+                currUserRef.remove();
+            }
+        } else {
+            // user not blocked
+            button.innerText = 'block user';
+            button.onclick = function() {
+                currUserRef.set(true);
+                // remove from each others' friend lists
+                const currFriendRef = firebase.database().ref('/users/'+currentUID+'/friends/'+uid);
+                const otherFriendRef = firebase.database().ref('/users/'+uid+'/friends/'+currentUID);
```

Comment:
Maybe you don't want to do this? It might give people a hint that they've been blocked which I think blocked functionalities don't like to do. Then again it could just be that the user unfriended you. I'll defer to your judgement and your other reviewers as long as it is considered.